packages/frontend/src/components/common/MantineModal/index.tsx

```diff
@@ -1,36 +1,81 @@
 import {
+    Button,
     Flex,
     type FlexProps,
     Group,
     Modal,
-    type ModalBaseSettings,
     type ModalBodyProps,
     type ModalContentProps,
     type ModalHeaderProps,
+    type ModalRootProps,
     Paper,
+    ScrollArea,
     Stack,
     Text,
-} from '@mantine/core';
+} from '@mantine-8/core';
 import { type Icon as IconType } from '@tabler/icons-react';
 import React from 'react';
 import MantineIcon from '../MantineIcon';
+import classes from './MantineModal.module.css';
 
-const modalSizes = {
+const MODAL_SIZES = {
     sm: 380,
-    lg: 480,
-    xl: 600,
+    md: 480,
+    lg: 560,
+    xl: 700,
     full: '100%',
+    auto: 'auto',
 } as const;
 
+type ModalSize = keyof typeof MODAL_SIZES;
+
 export type MantineModalProps = {
     opened: boolean;
     onClose: () => void;
     title: string;
     icon?: IconType;
-    size?: keyof typeof modalSizes;
-    children: React.ReactNode;
+    /**
+     * Modal size. Use predefined sizes ('sm', 'md', 'lg', 'xl', 'full', 'auto') or a custom number.
+     * @default 'lg'
+     */
+    size?: ModalSize | number;
```

Comment:
why can't we just simplify this and use `ModalRootProps['size']` instead? :thinking_face:

Reply:
gotcha, I think I know where you're going with this :eyes: